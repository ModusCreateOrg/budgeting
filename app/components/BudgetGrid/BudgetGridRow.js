import React from 'react';
import PropTypes from 'prop-types';

import formatAmount from 'utils/formatAmount';

import styles from './style.scss';

const BudgetGridRow = props => {
  const { transaction, categories } = props;
  const amount = formatAmount(transaction.value);
  const amountCls = amount.isNegative ? styles.neg : styles.pos;

  return (
    <tr key={transaction.id}>
      <td>
        <div className={styles.cellLabel}>Category</div>
        <div className={styles.cellContent}>
          {categories[transaction.categoryId]}
        </div>
      </td>
      <td>
        <div className={styles.cellLabel}>Description</div>
        <div className={styles.cellContent}>
          {transaction.description}
        </div>
      </td>
      <td className={amountCls}>
        <div className={styles.cellLabel}>Amount</div>
        <div className={styles.cellContent}>
          {amount.text}
        </div>
      </td>
    </tr>
  );
};
BudgetGridRow.propTypes = {
  transaction: PropTypes.object.isRequired,
  categories: PropTypes.object.isRequired,
};

export default BudgetGridRow;
