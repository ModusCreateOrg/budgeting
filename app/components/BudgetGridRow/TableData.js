// @flow
import * as React from 'react';
import { NavLink } from 'react-router-dom';
import styles from './style.scss';

type TableRowProps = {
  tdClass?: string,
  children: string,
  link: string,
};

const TableData = (props: TableRowProps) => {
  // Composing table rows into component in order to resolve the inablility of 'a' tags to wrap around 'td' tag
  const content = props.link ? (
    <NavLink
      to={props.link}
      activeStyle={{
        backgroundColor: '#ccc',
      }}
    >
      {props.children}
    </NavLink>
  ) : (
    <div className={styles.cellContent}>{props.children}</div>
  );

  return <td className={props.tdClass}>{content}</td>;
};

export default TableData;
