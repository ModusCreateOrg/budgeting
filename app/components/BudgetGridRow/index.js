// @flow
import * as React from 'react';
import formatAmount from 'utils/formatAmount';
import type { Transaction } from 'modules/transactions';
import type { Categories } from 'modules/categories';
import styles from './style.scss';
import NavLink from '../NavLink'
type BudgetGridRowProps = {
  transaction: Transaction,
  categories: Categories,
};

const BudgetGridRow = ({ transaction, categories }: BudgetGridRowProps) => {
  const amount = formatAmount(transaction.value);
  const amountCls = amount.isNegative ? styles.neg : styles.pos;
  const { id, categoryId, description } = transaction;
  const category = categories[categoryId];

  return (
    <tr key={id}>
      <td>
        <NavLink to={"/details/" + id} label={category} styles={styles}>
        </NavLink>
      </td>
      <td>
        <div className={styles.cellLabel}>Description</div>
        <div className={styles.cellContent}>{description}</div>
      </td>
      <td className={amountCls}>
        <div className={styles.cellLabel}>Amount</div>
        <div className={styles.cellContent}>{amount.text}</div>
      </td>
    </tr>
  );
};

export default BudgetGridRow;
