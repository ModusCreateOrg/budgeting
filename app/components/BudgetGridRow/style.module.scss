@import 'theme/variables';

.cellLabel {
  display: none;

  @media only screen and (max-width: $screen-small) {
    display: flex;
    align-items: center;
    flex: 1;
  }
}

.cellContent {
  flex: 1;
}

.neg {
  color: $red;
}

.pos {
  color: $green;
}
