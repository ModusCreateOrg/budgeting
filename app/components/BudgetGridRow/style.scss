@import 'theme/variables';

.navLink {
    &:first-child {
        display: none;
        @media only screen and (max-width: $screen-small) {
            display: flex;
            align-items: center;
            flex: 1;
        }
    }

    display: flex;
    flex: 1;
    text-decoration: none;
    color: $black;
    .neg & {
        color: $red;
    }
    .pos & {
        color: $green;
    }
}

.cellContent {
  flex: 1;
}