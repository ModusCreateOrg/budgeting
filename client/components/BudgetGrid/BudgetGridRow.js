import React, { PropTypes } from 'react';

import formatAmount from 'utils/formatAmount';

import styles from './style.scss';

const BudgetGridRow = (props) => {
  const { transaction, categories } = props;
  const amount = formatAmount(transaction.value);
  const amountCls = amount.isNegative ? styles.neg : styles.pos;

  return (
    <tr key={transaction.id}>
      <td>{categories[transaction.categoryId]}</td>
      <td>{transaction.description}</td>
      <td className={amountCls}>{amount.text}</td>
    </tr>
  );
};
BudgetGridRow.propTypes = {
  transaction: PropTypes.object.isRequired,
  categories: PropTypes.object.isRequired
};

export default BudgetGridRow;
