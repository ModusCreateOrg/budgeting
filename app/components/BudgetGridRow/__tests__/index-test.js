import React from 'react';
import renderer from 'react-test-renderer';
import BudgetGridRow from 'components/BudgetGridRow';

it('renders correctly', () => {
  const mockTransaction = {
    id: 1,
    description: "Trader Joe's food",
    value: -423.34,
    categoryId: 1,
  };

  const mockCategories = {
    1: 'Groceries',
    2: 'School',
  };

  const tree = renderer.create(<BudgetGridRow transaction={mockTransaction} categories={mockCategories} />).toJSON();
  expect(tree).toMatchSnapshot();
});
