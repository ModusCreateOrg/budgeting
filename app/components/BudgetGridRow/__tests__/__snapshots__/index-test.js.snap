// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<tr>
  <td>
    <a
      className="navLink "
      href="/details/1"
      onClick={[Function]}
    >
      Groceries
    </a>
  </td>
  <td>
    <div
      className="cellLabel"
    >
      Description
    </div>
    <div
      className="cellContent"
    >
      Trader Joe's food
    </div>
  </td>
  <td
    className="neg"
  >
    <div
      className="cellLabel"
    >
      Amount
    </div>
    <div
      className="cellContent"
    >
      -$423.34
    </div>
  </td>
</tr>
`;
